<a href="https://colab.research.google.com/github/justorfc/estadistica_2025_2/blob/main/Ejercicio_acero.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
%cd /content/drive/MyDrive/Ejercicio produccion de acero

/content/drive/MyDrive/Ejercicio produccion de acero


In [5]:
import os

os.chdir('/content/drive/MyDrive/Ejercicio produccion de acero')

In [6]:
os.getcwd()

'/content/drive/MyDrive/Ejercicio produccion de acero'

In [11]:
import pandas as pd

df_acero = pd.read_csv('acero.csv', decimal=',')
display(df_acero.head())

consumo  pr.tbc  pr.cc  pr.ca  pr.galv1  pr.galv2  pr.pint linea hora  \
0   135.31    6840    830      0       579      1401        0     A   1º   
1    84.08     443    903     58       611      1636      717     A   2º   
2   131.62    7270    572     36       982      1963      243     A   3º   
3    90.46    5031    694    122       896      1568        0     A   4º   
4   120.04    9365   1054    157       403      1480        0     A   5º   

  temperatura averias  naverias sistema  ProdTotal     NOx      CO     COV  \
0        Alta      Si         1     OFF      11266  0.4900  3.5450  0.5450   
1        Alta      No         0     OFF       7251  0.0725  2.8950  0.4250   
2        Baja      No         0     OFF      11066  1.4900  5.0075  0.6900   
3        Baja      No         0      ON       8311  1.7150  2.1600  0.3600   
4        Baja      No         0     OFF      12459  0.4650  4.8450  0.6625   

     SO2       CO2   N2O  
0  0.038  101.5000  6.35  
1  0.047   63.5650  2.23  
2  0.062   98.8175  5.99  
3  0.066   70.1825  3.66  
4  0.086   88.5300  6.06

In [12]:
df_acero.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 117 entries, 0 to 116
Data columns (total 20 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   consumo      117 non-null    float64
 1   pr.tbc       117 non-null    int64  
 2   pr.cc        117 non-null    int64  
 3   pr.ca        117 non-null    int64  
 4   pr.galv1     117 non-null    int64  
 5   pr.galv2     117 non-null    int64  
 6   pr.pint      117 non-null    int64  
 7   linea        117 non-null    object 
 8   hora         117 non-null    object 
 9   temperatura  117 non-null    object 
 10  averias      117 non-null    object 
 11  naverias     117 non-null    int64  
 12  sistema      117 non-null    object 
 13  ProdTotal    117 non-null    int64  
 14  NOx          117 non-null    float64
 15  CO           117 non-null    float64
 16  COV          117 non-null    float64
 17  SO2          117 non-null    float64
 18  CO2          117 non-null    float64
 19  N2O     

In [13]:
df_acero.describe()

consumo        pr.tbc        pr.cc       pr.ca    pr.galv1  \
count  117.000000    117.000000   117.000000  117.000000  117.000000   
mean   139.456496   7567.820513   295.529915  124.478632  402.803419   
std     55.185253   3002.703249   358.618556  161.990772  334.928923   
min     17.500000      0.000000     0.000000    0.000000    0.000000   
25%     99.090000   5950.000000     0.000000    0.000000    0.000000   
50%    140.070000   8808.000000    80.000000    0.000000  432.000000   
75%    182.480000   9734.000000   582.000000  248.000000  698.000000   
max    290.720000  10979.000000  1204.000000  677.000000  982.000000   

          pr.galv2     pr.pint    naverias     ProdTotal         NOx  \
count   117.000000  117.000000  117.000000    117.000000  117.000000   
mean   1159.717949  188.555556    0.675214  11841.649573    1.196132   
std     577.791602  289.445857    1.292078   4075.068276    0.643837   
min       0.000000    0.000000    0.000000   2187.000000    0.012500   
25%     933.000000    0.000000    0.000000   9023.000000    0.775000   
50%    1360.000000    0.000000    0.000000  11860.000000    1.120000   
75%    1567.000000  394.000000    0.000000  14908.000000    1.655000   
max    1963.000000  898.000000    4.000000  23202.000000    2.775000   

               CO         COV         SO2         CO2         N2O  
count  117.000000  117.000000  117.000000  117.000000  117.000000  
mean     4.377350    0.620983    0.066504  104.628141    6.114786  
std      2.095038    0.255447    0.022064   41.389709    2.125250  
min      0.250000    0.025000    0.001000   14.285000    0.870000  
25%      2.927500    0.452500    0.054000   73.717500    4.710000  
50%      4.107500    0.625000    0.066000  105.307500    6.170000  
75%      5.585000    0.785000    0.080000  134.877500    7.580000  
max     10.170000    1.255000    0.127000  218.307500   11.140000

La salida de `df_acero.describe()` proporciona estadísticas descriptivas de las columnas numéricas en tu DataFrame. Aquí te explico cada una de ellas:

*   **count**: Muestra el número de valores no nulos en cada columna. En este caso, todas las columnas numéricas tienen 117 valores no nulos, lo que indica que no hay datos faltantes en estas columnas.
*   **mean**: Es el promedio de los valores en cada columna. Por ejemplo, el promedio de `consumo` es aproximadamente 139.46.
*   **std**: Es la desviación estándar, que mide la dispersión de los datos alrededor de la media. Una desviación estándar alta indica que los datos están más dispersos.
*   **min**: Es el valor mínimo en cada columna.
*   **25% (Q1)**: Es el primer cuartil, que representa el percentil 25. El 25% de los datos son menores o iguales a este valor.
*   **50% (median)**: Es la mediana, que representa el percentil 50. Es el valor central cuando los datos están ordenados.
*   **75% (Q3)**: Es el tercer cuartil, que representa el percentil 75. El 75% de los datos son menores o iguales a este valor.
*   **max**: Es el valor máximo en cada columna.

Esta tabla te da un resumen rápido de la distribución y rango de los datos en cada una de las columnas numéricas.